packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/Providers/Abstractions/HttpMessageApi.cs

```diff
@@ -0,0 +1,36 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Microsoft.Generator.CSharp.Expressions;
+using Microsoft.Generator.CSharp.Primitives;
+using Microsoft.Generator.CSharp.Snippets;
+using Microsoft.Generator.CSharp.Statements;
+
+namespace Microsoft.Generator.CSharp.ClientModel.Providers
+{
+    public abstract record HttpMessageApi : ScopedApi, IHttpMessageApi
+    {
+        protected HttpMessageApi(Type type, ValueExpression original) : base(type, original)
+        {
+        }
+
+        public abstract HttpRequestApi Request();
```

Comment:
Do we want to somehow distinguish between members that produce properties from members that produce methods? I think it will be kind of redundant to suffix all of these with Property, but I was a little confused at first as to what this represents.

Reply:
our expression system does not have this ability unless we add suffixes.

Reply:
Is it possible the default implementation is from method, but sub-plugin is from property?